bitcoin/src/amount.rs

```diff
@@ -83,8 +83,28 @@ impl Denomination {
             Denomination::MilliSatoshi => "msat",
         }
     }
+
+    /// The different str forms of denominations that are recognized.
+    fn forms(s: &str) -> Option<Self> {
+        match s {
+            "BTC" | "btc" | "BTCs" | "btcs" => Some(Denomination::Bitcoin),
+            "cBTC" | "cbtc" | "cBTCs" | "cbtcs" => Some(Denomination::CentiBitcoin),
+            "mBTC" | "mbtc" | "mBTCs" | "mbtcs" => Some(Denomination::MilliBitcoin),
+            "uBTC" | "ubtc" | "uBTCs" | "ubtcs" => Some(Denomination::MicroBitcoin),
+            "nBTC" | "nbtc" | "nBTCs" | "nbtcs" => Some(Denomination::NanoBitcoin),
+            "pBTC" | "pbtc" | "pBTCs" | "pbtcs" => Some(Denomination::PicoBitcoin),
+            "bit" | "bits" => Some(Denomination::Bit),
+            "SATOSHI" | "Satoshi" | "Satoshis" | "satoshi" | "satoshis" | "Sat" | "sat" | "SAT"
+            | "sats" => Some(Denomination::Satoshi),
+            "mSAT" | "msat" | "mSATs" | "msats" => Some(Denomination::MilliSatoshi),
```

Comment:
```suggestion
            "SATOSHI" | "satoshi" | "SATOSHIS" | "satoshis" | "SAT" | "sat" | "SATS" | "sats"
            => Some(Denomination::Satoshi),
            "mSAT" | "mSATS" | "msat" |  "msats" => Some(Denomination::MilliSatoshi),
```

Reply:
I.e caps or non-caps, plural or non-plural. Never first character capital and the rest lower - dollars/cents is never capitalised, neither should bitcoin or sats be (Bitcoin is the network, bitcoin is the unit).

Reply:
done :+1: